packages/plugins/starttls/index.js

```diff
@@ -13,26 +13,25 @@ const net = require('net')
 const NS = 'urn:ietf:params:xml:ns:xmpp-tls'
 
 function match(features, entity) {
-  return features.getChild('starttls', NS) &&
-    entity.socket.Socket === net.Socket // https://prosody.im/issues/issue/837
```

Comment:
I'm not really sure what this check trying to accomplish, as `entity.socket.Socket` is `undefined` always when I'm tested.

`entity.socket` is an instance of `net.Socket`, so we might want to ask:
`entity.socket instanceof net.Socket` ( will be also `true` for `TLSSocket` ) or `entity.socket.constructor ===  net.Socket` ( only `true` for `net.Socket` )
( i'll commit the later, as it seems to be most relevant to the original check )